File: joiner-test-support/src/main/java/cz/encircled/joiner/TestDataListener.java
package cz.encircled.joiner;

import org.springframework.test.context.TestContext;
import org.springframework.test.context.TestExecutionListener;

/**
 * @author Kisel on 26.01.2016.
 */
public class TestDataListener implements TestExecutionListener {

    public void beforeTestClass(TestContext testContext) throws Exception {

    }

    public void prepareTestInstance(TestContext testContext) throws Exception {
        TestData bean = testContext.getApplicationContext().getBean(TestData.class);
        bean.prepareData();
    }

    public void beforeTestMethod(TestContext testContext) throws Exception {

    }

    public void afterTestMethod(TestContext testContext) throws Exception {

    }

    public void afterTestClass(TestContext testContext) throws Exception {

    }
}

File: joiner-core/src/test/java/cz/encircled/joiner/test/core/data/TestDataListener.kt
package cz.encircled.joiner.test.core.data

import org.springframework.test.context.TestContext
import org.springframework.test.context.TestExecutionListener

/**
 * @author Kisel on 26.01.2016.
 */
class TestDataListener : TestExecutionListener {

    @Throws(Exception::class)
    override fun beforeTestClass(testContext: TestContext) {

    }

    @Throws(Exception::class)
    override fun prepareTestInstance(testContext: TestContext) {
        val bean = testContext.applicationContext.getBean(TestData::class.java)
        bean.prepareData()
    }

    @Throws(Exception::class)
    override fun beforeTestMethod(testContext: TestContext) {

    }

    @Throws(Exception::class)
    override fun afterTestMethod(testContext: TestContext) {

    }

    @Throws(Exception::class)
    override fun afterTestClass(testContext: TestContext) {

    }
}
